src/components/RegisterForm.vue

```diff
@@ -0,0 +1,190 @@
+    <div v-if="genderValidation">
+    <select v-model="gender" class="Valid">
+      <option disabled value=""></option>
+      <option value="Male">Male</option>
+      <option value="Female">Female</option>
+      <option value="Both">Both</option>
+    </select></div>
+    <div v-else>*gender has to be selected*<br><br>
+    <select v-model="gender">
+      <option disabled value=""></option>
+      <option value="Male">Male</option>
+      <option value="Female">Female</option>
+      <option value="Both">Both</option>
+    </select></div>
+
+    <h2>Your Avatar:</h2>
+    <label v-if="imgNoValidation">*please select an Avatar*<br><br></label>
+    <label v-for="img in imgs" :key="img.src">
+      <input type="radio" v-model="image" :value="img.value">
+      <img :src="img.src" alt="avatar" /></label><br><br>
+
+    <button @click.prevent="submit"> Submit</button> <br><br> 
+    <hr size="3" noshade>
+  
+  </div>
+</template>
+
+<script>
+export default {
+  data: function() {
+    return {
+      fname: '',
+      mail: '',
+      phonenum: '',
+      gender: '',
+      image: '',
+      msg: 'Register Here',
+      imgs:[
+        {src: require("../assets/icons/dog.png") , value:"dog"},
+        {src: require("../assets/icons/fish.png"), value:"fish"},
+        {src: require("../assets/icons/owl.png"), value:"owl"}
+      ]
+    };
+  },
+  methods: {
+    submit: function(){
+      if(this.emailValidation && this.phonenumValidation&&
+         this.fnameValidation && this.genderValidation&&
+         !this.imgNoValidation)
+      {
+        alert('submitted');
+        console.log(this.jsonOutput);
+        this.cleanForm()
+      }
+      else{
+        alert('oops');
+      }
+    },
+    cleanForm: function(){
+      this.fname = '';
+      this.mail = '';
+      this.phonenum = '';
+      this.gender = '';
+      this.image = '';
+    }
+  },
+  computed: {
+    // fnameCaps: function() {
+    //   return this.fname.toUpperCase();
+    fnameValidation: function(){
+      if(this.fname.match(/^([a-zA-Z]{1,} )([a-zA-Z]{1,})$/))
+        return true;
+      else
+        return false;
+    },
+    emailValidation: function () {
+      if (this.mail.match(/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/)) {
+        return true;
+      }
+      else{
+       return false;
+      }
+    },
+    phonenumValidation: function(){
+      if (this.phonenum.match(/^\d{10}$/)) {
+        return true;
+      }
+      return false;
+    },
+    genderValidation: function(){
+      if(this.gender != '')
+        return true;
+      else
+        return false;
+    },
+    imgNoValidation: function(){
+      if(this.image == '')
+        return true;
+      else
+        return false;
+    },
+    jsonOutput: function(){
+      return [
+        ['fname', this.fname],
+        ['mail', this.mail],
+        ['phonenum', this.phonenum],
+        ['gender', this.gender],
+        ['image', this.image]
+      ];
+    },
+    emptyFname: function(){
```

Comment:
in general the code handling name/mail/phone is almost identical not only in the HTML but also the JS part - I think you can probably remove a lot of duplicated code by handling them together (for example comp-prop that return an object of validation of everything instead of a comp-prop for each one - than you can probably iterate through the fields in a loop instead of writing the same lines again and again)